src/TLuaInterpreter.cpp

```diff
@@ -10483,81 +10531,81 @@ int TLuaInterpreter::echoPopup(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#echoLink
 int TLuaInterpreter::echoLink(lua_State* L)
 {
-    string a1;
-    string a2;
-    string a3;
-    string a4;
-    bool a5 = false;
+    QString a1;
+    QString a2;
+    QString a3;
+    QString a4;
+    bool useCurrentFormatElseDefault = false;
     bool gotBool = false;
 
     int s = 1;
     int n = lua_gettop(L);
 
     if (!lua_isstring(L, s)) {
-        lua_pushstring(L, "echoLink: wrong argument type");
+        lua_pushfstring(L, "echoLink: bad argument #%d type (string expected, got %s!)", s, luaL_typename(L, s));
```

Comment:
This is one function where I haven't worked out what the arguments are - might be worth checking what is said in the Wiki - if that says anything more that can be put here...

Reply:
Echolink is a function that will accept two 'optional' arguments, the one at the beginning if given a window name. and the last argument is a useCurrentFormatElseDefault. So I'm working with this tricky case here.

Reply:
I'm going to work its logic so it can be clear and know which error message to deliver

Reply:
SLySven, this feedback has been applied in other patch. :+1:

Reply:
echolinks fixes are in. applied.